static/app/components/events/interfaces/spans/spanProfileDetails.tsx

```diff
@@ -0,0 +1,194 @@
+import {useMemo} from 'react';
+
+import StackTrace from 'sentry/components/events/interfaces/crashContent/stackTrace';
+import {EventTransaction, Frame, PlatformType} from 'sentry/types/event';
+import {STACK_VIEW} from 'sentry/types/stacktrace';
+import {defined} from 'sentry/utils';
+import {CallTreeNode} from 'sentry/utils/profiling/callTreeNode';
+import {Frame as ProfilingFrame} from 'sentry/utils/profiling/frame';
+import {Profile} from 'sentry/utils/profiling/profile/profile';
+import {formatTo} from 'sentry/utils/profiling/units/units';
+import {useProfileGroup} from 'sentry/views/profiling/profileGroupProvider';
+
+import {SpanType} from './types';
+
+interface SpanProfileDetailsProps {
+  event: Readonly<EventTransaction>;
+  span: Readonly<SpanType>;
+}
+
+export function SpanProfileDetails({event, span}: SpanProfileDetailsProps) {
+  const profileGroup = useProfileGroup();
+
+  // TODO: Pick another thread if it's more relevant.
+  const threadId = useMemo(
+    () => profileGroup.profiles[profileGroup.activeProfileIndex]?.threadId,
+    [profileGroup]
+  );
+
+  const profile = useMemo(() => {
+    if (!defined(threadId)) {
+      return null;
+    }
+    return profileGroup.profiles.find(p => p.threadId === threadId) ?? null;
+  }, [profileGroup.profiles, threadId]);
+
+  const nodes: CallTreeNode[] = useMemo(() => {
+    if (profile === null) {
+      return [];
+    }
+
+    const relativeStartTimestamp = formatTo(
+      span.start_timestamp - event.startTimestamp,
+      'second',
+      profile.unit
+    );
+    const relativeStopTimestamp = formatTo(
+      span.timestamp - event.startTimestamp,
+      'second',
+      profile.unit
+    );
+
+    return getTopNodes(profile, relativeStartTimestamp, relativeStopTimestamp);
+  }, [profile, span, event]);
+
+  const frames = useMemo(() => {
+    if (!nodes.length) {
+      return [];
+    }
+
+    return extractFrames(nodes[0], event.platform || 'other');
+  }, [event, nodes]);
+
+  if (!defined(profile)) {
+    return null;
+  }
+
+  if (!nodes.length) {
+    return null;
+  }
+
+  return (
+    <StackTrace
+      event={event}
+      hasHierarchicalGrouping
+      newestFirst
+      platform={event.platform || 'other'}
+      stacktrace={{
+        framesOmitted: null,
+        hasSystemFrames: false,
+        registers: null,
+        frames,
+      }}
+      nativeV2
+      stackView={STACK_VIEW.APP}
+    />
+  );
+}
+
+function getTopNodes(profile: Profile, startTimestamp, stopTimestamp): CallTreeNode[] {
+  let duration = 0;
+
+  const callTree: CallTreeNode = new CallTreeNode(ProfilingFrame.Root, null);
+
+  for (const sample of profile.samples) {
```

Comment:
Feels like we should just be adding a .timestamp to each sample when we construct the chart so you could skip iterating over a ton of samples that way

Reply:
I like that idea, something like this would make for a unified interface for aligning profiles + transactions.